### ISSUE TITLE
[Bug]: search-box component is not adding title, aria-lable, aria-labelledby to the parent div of input element whereas it is adding only role to the parent div #

### ISSUE THREAD

**Issue by Govindu9** [microsoft/fluentui#29555]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
PS C:\source\M365AdminUX> npx envinfo --system --npmPackages '{@fluent/*}' --browsers
npx: installed 1 in 4.323s

  System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i5-10310U CPU @ 1.70GHz
    Memory: 4.07 GB / 15.85 GB
  Browsers:
    Edge: Spartan (44.22621.2283.0), Chromium (116.0.1938.81)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/mohan-sri-the-sasster/pen/RwEjRXR

## Actual Behavior
fill this out
title and ariaLabel attributes are not added to the parent div of input element. Only role is getting added to parent div of input element
## Expected Behavior
fill this out
Like how role attribute is added to parent div of input element the same should happen with title , ariaLable props as well.

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by behowell** [comment]
Hi @Govindu9, SearchBox seems to be working correctly here. Native props like `ariaLabel` that are added the SearchBox control get forwarded to the `<input>` inside. That input already has `role="searchbox"` (without you needing to add the role to the SearchBox), and is the correct element to get `aria-label` so that screen readers read the label when focus is on the SearchBox.

Looking at the code for SearchBox, it doesn't look like there is a way for you to add props to its wrapper `<div>` element. The one exception is the `role` prop (as you have found). However, you should _not_ be setting `role="searchbox"` on the wrapper, since that role is already assigned to the `<input>` inside. From the [SearchBox documentation](https://developer.microsoft.com/en-us/fluentui#/controls/web/searchbox), the `role` prop on SearchBox itself is used for:
> The role assigned to the root DIV element of the SearchBox, useful for defining a landmark role, such as "search".

In summary, if you remove `role="searchbox"` from the code in your example, then it should be doing the right thing:
```diff
-<SearchBox placeholder="Search" role="searchbox" title="search" ariaLabel="searchbox" />
+<SearchBox placeholder="Search" title="search" ariaLabel="searchbox" />
```

Please let me know if I've misunderstood the issue. Thanks.

**Comment by Govindu9** [comment]
FYI.. 
@behowell 

we cannot fix this issue. it is issue from fluent UI. 


![image](https://github.com/microsoft/fluentui/assets/145434269/b8e862ca-5527-40b9-b49f-84f48482766d)

we can fix the above bug only by adding either of aria-label, title, aria-labelledBy attributes. 

Here when we are passing attributes like ariaLabel, title, aria-labelledBy it is getting applied to the input element only and not to the parent div. 
 
Problem :  only  Role attribute is getting applied to the parent div.  where as ariaLabel, aria-labelledBy,title are applied to the input tag only. that is causing the issue . 

Image


This needs to fixed from fluentUI.

**Comment by behowell** [comment]
Hi @Govindu9, the input is the correct element to have `aria-label`, etc. attributes. Like I mentioned in my previous reply, the fix for the issue you're seeing is to _remove_ "`role="searchbox"` from the SearchBox control. The `<input>` inside the SearchBox automatically has that role, so you shouldn't add it to the SearchBox's root div.

Let me know if that resolves the issue you're seeing.

**Comment by Govindu9** [comment]
@behowell 
We have scenario were searchbox is placed inside a menubar . so we need to pass role as menuItem which is not getting applied to the <input> that  is causing the issue.  
Please find the below snip shot for more info

![image](https://github.com/microsoft/fluentui/assets/145434269/3219aaec-dce1-4b44-a1be-0efea3d76524)

**Comment by behowell** [comment]
@Govindu9 that error is telling you that you can't have a searchbox inside a menubar. You'll need to modify your design so the searchbox is not a child of the menubar, rather than changing the role of the searchbox. See the [documentation for the menubar role on MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menubar_role) for more info about what is a valid child of a menubar.

Closing this as it is not an issue in FluentUI.
